src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HPackEncoder.cs

```diff
@@ -106,7 +106,7 @@ public static bool EncodeLiteralHeaderFieldWithoutIndexing(int index, Span<byte>
         }
 
         /// <summary>Encodes a "Literal Header Field without Indexing - New Name".</summary>
-        public static bool EncodeLiteralHeaderFieldWithoutIndexingNewName(string name, string value, Span<byte> destination, out int bytesWritten)
+        public static bool EncodeLiteralHeaderFieldWithoutIndexingNewName(string name, string[] values, string separator, Span<byte> destination, out int bytesWritten)
```

Comment:
Is this new parameter `separator` ever going to take on a value other than `HttpHeaderParser.DefaultSeparator`? If not, it seems unnecessary.